x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -264,7 +264,7 @@ private boolean checkIndexMappingUpToDate(ClusterState clusterState) {
          *
          * ...which is true if the mappings have been updated.
          */
-        return checkIndexMappingVersionMatches(clusterState, descriptor.getMappingVersion()::onOrBefore);
+        return checkIndexMappingVersionMatches(clusterState, descriptor.getMappingsNodeVersion()::onOrBefore);
```

Comment:
Leaving this here because this is the most relevant one for me: What's the expected timeline for moving away from this deprecated method and moving over the the version field entirely?

Reply:
I should be able to do that in the next week or so. There's a follow-up PR coming soon that will make these new versions available via cluster state, and once that's merged, we'll be able to calculate the minimum mappings version for each system index descriptor in the cluster. Once that's in, I'll update the "prior system index descriptor" mechanism (https://github.com/elastic/elasticsearch/issues/98505), and that will change the `getDescriptorCompatibleWith` method to use the mappings version rather than the node version.